x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/rule_definition_section.tsx

```diff
@@ -109,7 +114,7 @@ export const Filters = ({
       responsive={false}
       gutterSize="xs"
     >
-      <FilterItems filters={flattenedFilters} indexPatterns={[indexPattern as DataView]} readOnly />
+      {dataView && <FilterItems filters={flattenedFilters} indexPatterns={[dataView]} readOnly />}
```

Comment:
Can `dataView` really be undefined? Or is this just to please TS?

If it can, then the whole component should be rendered conditionally, early returning, and the existing JSX remain unchanged:
```ts
if(!dataView) {
  return null;
}
```

Reply:
Yes, it was to make TS happy, as dataView can be undefined. 
You are right, I will implement your suggestion.